scripts/Crawlers/BaseCrawler.py

```diff
@@ -0,0 +1,209 @@
+from datalad import api
+import git
+import os
+import re
+import requests
+import abc
+import json
+
+
+class BaseCrawler:
```

Comment:
Add a docstring to the class explaining the sequence along which the 3 abstract methods will be called (how they are related).

Reply:
Summarize clearly the steps required to create a new crawler.